utils/upload.go

```diff
@@ -0,0 +1,180 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+
+	"cloud.google.com/go/storage"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+)
+
+// channel to extract files from the folder
+type fileWalk chan string
+
+type Uploader interface {
+	Upload(walker fileWalk)
+}
+
+func UploadToCloudStorage(uploader Uploader, path string) {
+	walker := make(fileWalk)
+	go func() {
+		//get files to upload via the channel
+		if err := filepath.Walk(path, walker.WalkFunc); err != nil {
+			log.Fatalln("Walk failed: ", err)
+		}
+
+		close(walker)
+
+	}()
+
+	uploader.Upload(walker)
+
+}
+
+func (f fileWalk) WalkFunc(path string, info os.FileInfo, err error) error {
+	if err != nil {
+		return err
+	}
+
+	if !info.IsDir() {
+		f <- path
+	}
+
+	return nil
+}
+
+type AwsUploader struct {
+	BucketName      string
+	Prefix          string
+	Region          string
+	FolderLocalPath string
+}
+
+func (a AwsUploader) Upload(walker fileWalk) {
+
+	//creating a new session
+	sess, err := session.NewSession(&aws.Config{
+		Region:                        aws.String(a.Region),
+		CredentialsChainVerboseErrors: aws.Bool(true),
+	})
+	if err != nil {
+		log.Fatalf("failed to create a session")
+	}
+
+	bucket := a.BucketName
+	log.Printf("bucket %s", bucket)
+
+	prefix := a.Prefix
+
+	uploader := s3manager.NewUploader(sess)
+	for path := range walker {
+		// rel, err := filepath.Rel(a.FolderLocalPath, path)
+		// if err != nil {
+		// 	log.Fatalln("Unable to get relative path ", a.FolderLocalPath, path)
+		// }
+		filename := filepath.Base(path)
+
+		file, err := os.Open(path)
+		if err != nil {
+			log.Println("Failed opening file", path, err)
+			continue
+		}
+
+		defer file.Close()
+
+		result, err := uploader.Upload(&s3manager.UploadInput{
+			Bucket: &bucket,
```

Comment:
You can use `Bucket: aws.String(constants.S3_BUCKET_NAME)`

Reply:
do you want me always use the bucket name in constants?

Reply:
Yep!

Reply:
I mean you can create a new constant for GCP and Azure bucket name as well